From 6ff1f82eee19259e55ceb949678e33c8476e291f Mon Sep 17 00:00:00 2001
From: Sivashankari Madhavan <quic_sivamadh@quicinc.com>
Date: Fri, 9 Jun 2023 09:50:45 +0530
Subject: [PATCH] hostapd: Fix the country code change bring up failure issue

While changing the country, randomly hostapd initiate the regulatory
changed and process the SKIP CAC by user before the driver list update
for a changed country. So that the hostapd processing the old
channel list DFS state and skipping the CAC.

Fix it by avoiding the regulatory change update if self managed
regulatory support enabled.

Signed-off-by: Sivashankari Madhavan <quic_sivamadh@quicinc.com>
---
 src/drivers/driver_nl80211_event.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/src/drivers/driver_nl80211_event.c
+++ b/src/drivers/driver_nl80211_event.c
@@ -4160,6 +4160,10 @@ static void do_process_drv_event(struct
 		break;
 	case NL80211_CMD_REG_CHANGE:
 	case NL80211_CMD_WIPHY_REG_CHANGE:
+		if (cmd == NL80211_CMD_REG_CHANGE &&
+		   drv->capa.flags & WPA_DRIVER_FLAGS_SELF_MANAGED_REGULATORY) {
+			break;
+		}
 		nl80211_reg_change_event(drv, tb);
 		break;
 	case NL80211_CMD_REG_BEACON_HINT:
